api/actions/app_logs.go

```diff
@@ -36,14 +37,17 @@ func (a *AppLogs) Read(ctx context.Context, logger logr.Logger, authInfo authori
 		return nil, apierrors.LogAndReturn(logger, apierrors.ForbiddenAsNotFound(err), "Failed to fetch app from Kubernetes", "AppGUID", appGUID)
 	}
 
+	var buildLogs []repositories.LogRecord
 	build, err := a.buildRepo.GetLatestBuildByAppGUID(ctx, authInfo, app.SpaceGUID, appGUID)
 	if err != nil {
-		return nil, apierrors.LogAndReturn(logger, apierrors.ForbiddenAsNotFound(err), "Failed to fetch latest app CFBuild from Kubernetes", "AppGUID", appGUID)
-	}
-
-	buildLogs, err := a.buildRepo.GetBuildLogs(ctx, authInfo, app.SpaceGUID, build.GUID)
-	if err != nil {
-		return nil, apierrors.LogAndReturn(logger, err, "Failed to fetch build logs", "AppGUID", appGUID, "BuildGUID", build.GUID)
+		if !errors.As(err, new(apierrors.NotFoundError)) {
+			return nil, apierrors.LogAndReturn(logger, apierrors.ForbiddenAsNotFound(err), "Failed to fetch latest app CFBuild from Kubernetes", "AppGUID", appGUID)
+		}
+	} else {
+		buildLogs, err = a.buildRepo.GetBuildLogs(ctx, authInfo, app.SpaceGUID, build.GUID)
```

Comment:
If there isn't a build for the given app GUID yet, I think that it is safe to assume that there won't be any logs either. Also, the `build.GUID` parameter to the call to `GetBuildLogs` will be empty.

You could probably simply return an empty `repositories.LogRecord` array and a nil error inside the check for an error from `GetLatestBuildByAppGUID` (lines 42-46).

Reply:
The GetBuildLogs command only runs if there is no error from GetLatestBuildByAppGUID (Including not found errors) so it defaults to the var buildLogs which is an empty list that will eventually get merged with the result of GetRuntimeLogsForApp below. If we're sure that there will be never be any app logs from GetRuntimeLogsForApp then we can return an empty list and no err here instead.

Reply:
Thanks for the clarification. While that does make sense now, I think it is more Golang-idiomatic to short-circuit and handle the empty list return inside the non-nil error check so that you know that the code that runs after that does have a build to work with. Personally, I find that style easier to read/understand than using else statements.

Reply:
I think I understand. Since there is no build there won't be an app so we can just return an empty list immediately instead of querying the app logs and merging the empty build log list. Fixed